### PR TITLE
feat: make data sync interval configurable

### PR DESCRIPTION
**Pull request by dejanzele**

### CODE REVIEW

File: internal/yunikorn/service.go [hunk 1]
```diff
@@ -29,7 +29,15 @@ type Service struct {
...
-func NewService(ctx context.Context, repository repository.Repository, eventRepository repository.EventRepository, client Client) *Service {
+type Option func(*Service)
+
+func WithSyncInterval(interval time.Duration) Option {
+	return func(s *Service) {
+		s.syncInterval = interval
+	}
+}
+
+func NewService(repository repository.Repository, eventRepository repository.EventRepository, client Client, opts ...Option) *Service {
```
Comment (richscott) [inline internal/yunikorn/service.go:40]: `ctx` is deleted - do/will we need a Context for doing cancellation, etc, in the future?

**Review by richscott — COMMENTED** [review]
Just one question about removing a Context in the NewService() constructor.

File: internal/yunikorn/service.go (hunk 1 above)
Comment (dejanzele) [reply internal/yunikorn/service.go:40]: the context was actually unused, it got changed now to start a goroutine during instantiate, so it was just sitting idle

Comment (richscott) [reply internal/yunikorn/service.go:40]: 👍🏼

**Review by richscott — APPROVED** [review]

**Pull request merged**